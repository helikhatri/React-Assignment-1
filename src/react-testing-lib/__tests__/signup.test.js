import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow,mount } from 'enzyme';
import Signup from '../../containers/Signup'
import Button from "react-bootstrap/Button";
import Form from "react-bootstrap/Form";
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });
let wrapper;

beforeEach(() => {
    wrapper = mount(<Signup />);
});

describe('Signup', () => {
    it('has a Signup button', () => {
        expect(wrapper.containsMatchingElement(<Button type='submit'>Signup</Button>))
    })

    it('has a email input field', () => {
        expect(wrapper.containsMatchingElement(<Form.Control type='email'>Email</Form.Control>))
    })

    it('has a password input field', () => {
        expect(wrapper.containsMatchingElement(<Form.Control type='password'>Password</Form.Control>))
    })

    it('snapshot test for signup', () => {
        const tree = renderer
        .create(<Signup>signup</Signup>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
})