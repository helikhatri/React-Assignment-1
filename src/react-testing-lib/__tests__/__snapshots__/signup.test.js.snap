// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup snapshot test for signup 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="form-group"
    size="lg"
  >
    <label
      className="form-label"
      htmlFor="email"
    >
      Email
    </label>
    <input
      autoFocus={true}
      className="form-control"
      id="email"
      onChange={[Function]}
      type="email"
      value=""
    />
  </div>
  <div
    className="form-group"
    size="lg"
  >
    <label
      className="form-label"
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="form-control"
      id="password"
      onChange={[Function]}
      type="password"
      value=""
    />
  </div>
  <div
    className="form-group"
    size="lg"
  >
    <label
      className="form-label"
      htmlFor="confirmPassword"
    >
      Confirm Password
    </label>
    <input
      className="form-control"
      id="confirmPassword"
      onChange={[Function]}
      type="password"
      value=""
    />
  </div>
  <button
    className="btn btn-success btn-lg"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
    Signup
  </button>
</form>
`;
